proposals/infra/INF-0005-Long-Vector-TestPlan.md

```diff
@@ -0,0 +1,254 @@
+covered by existing test collateral. But, as part of this work we will verify
+that assumption.
+
+# Implementation phases
+Do the test work in two simple phases.
+
+1. Implement and validate (locally against WARP) the 3 test categories.
+2. HLK related work:
+    - Add a SM 6.9 HLK requirement. Includes updating the HLK requirements doc.
+    - Update mm_annotate_shader_op_arith_table.py to annotate the new test cases
+    with HLK GUIDS and requirements
+    - Add new tests to HLK playlist
+
+# Shipping
+Note that because DXC and the Agility SDK are both undocked from Windows it is
+our normal operating behavior for the HLK tests to become available with a later
+TBD OS release. The good news is that this doesn't prevent the test from being
+available much earlier in the DXC repo. Just that they are simply TAEF tests in
+the DXC repo. An HLK test includes an extra level of infrastructure for test
+selection and result submission for WHQL signing of drivers.
+
+1. Tests will be shared privately with IHV's along with the latest DXC and
+latest Agility SDK for tesing and validation. IHVs will also be able to build
+and run the tests from the public DXC repo themselves.
+
+2. The tests will ship with the HLK at a TBD date in a later OS release.
+
+# New HLK Tests
+mm_annotate_shader_op_arith_table.py (WinTools repo) will need to be updated to
+recognize any new tests (not test cases) added. And additional GUIDs added for
+new test cases. mm_annotate_shader_op_arith_table.py is called by
+mm-hlk-update.py when converting from 'ExecTests' to the HLK 'DxilConf' tests.
+The aforementioned *table.py script is run by Integration.HLKTestsUpdate.yaml
+
+# Test Validation Requirements
+The following statements must be true and validated for this work to be
+considered completed.
+- All new test cases pass when run locally against a WARP device
+- All new test cases are confirmed to be lit up and available in the HLK when
+  the target device reports support
+- All new tests/test cases are added to the HLK playlist
+
+# Requirements
+     - Greg's long vector changes:
+       https://github.com/microsoft/DirectXShaderCompiler/pull/7188
+     - WARP long vector support (Jesse). Needs Greg's work, or a private WIP
+        branch of Greg's work. ETA of ~1 week to implement.
+
+# Notes
+- The 'ExecTests' (ExecutionTest.cpp) are used to generate the HLK tests
+   (DxilConf tests)
+- Private test binaries/collateral will be shared with IHVs for validation
+   purposes. This will enable IHVs to verify long vector functionality without
+   waiting for an OS/HLK release.
+
+# Open Questions
+- We speculate that the combination of test cases for HLSL operators and HLSL
+intrinsics which map to DXIL ops should get us all, or most of the coverage
+required for the 'emulated' intrinsics. But we should audit that.
+
+# HLSL Operators
+These operators should generate LLVM native ops which use vectors.
```

Comment:
It might be worth digging a bit deeper here into which actual LLVM operations you want to be testing.

For example, I don't think that the `,` operator actually corresponds to anything the driver would see?

Reply:
I had naively checked what worked on existing vectors in godbolt. Updating the table based on renewed understanding and these comments.

Reply:
@pow2clk I think it would be helpful for me to discuss casts a little more offline. And I can update notes here based on that.